client/alloc_watcher.go

```diff
@@ -463,6 +463,9 @@ func (p *remotePrevAlloc) streamAllocDir(ctx context.Context, resp io.ReadCloser
 		}
 	}
 
+	// if we see this file, there was an error on the remote side
+	errorFilename := allocdir.SnapshotErrorFilename(p.prevAllocID)
```

Comment:
I might be missing something obvious, but I can't find where allocdir is initialized. Can it ever be nil?

Reply:
`allocdir` is sneakily referring to a package here... kind of unfortunate I use `allocDir` a lot as a variable and `allocdir` a lot as a package... :grimacing: